designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessStateDefinitionManager.scala

```diff
@@ -8,14 +8,42 @@ import pl.touk.nussknacker.engine.deployment.ExternalDeploymentId
 import java.net.URI
 
 //@TODO: In future clean up it.
+/**
+  * Provides state definitions, state transitions and status property transformations for UI visualization.
+  */
 trait ProcessStateDefinitionManager {
+
+  /**
+    * Default set of state definitions that provides state defaults
+    * and allows filtering by fixed set of status IDs (see [[ProcessState]] and [[StateStatus.name]]).
+    */
+  def stateDefinitions(): Set[StateDefinition]
```

Comment:
Maybe it is better to keep definitions as `Map[StatusName, StateDefinition]` to enforce key-by-name uniquness.

Reply:
I aggree